client/client.go

```diff
@@ -416,6 +473,33 @@ func (d *dynamicWatcher) AddOrUpdateWatcher(watcher ObjectIdentifier, watchedObj
 	return nil
 }
 
+// watchLatest performs a list with the field selector for the input watched object to get the resource version and then
+// starts the watch using the client-go RetryWatcher API. The returned bool indicates that the input watched object
+// exists on the cluster.
+func watchLatest(watchedObject ObjectIdentifier, resource dynamic.ResourceInterface) (apiWatch.Interface, bool, error) {
+	fieldSelector := "metadata.name=" + watchedObject.Name
+	timeout := int64(10)
+
+	listResult, err := resource.List(
+		context.TODO(), metav1.ListOptions{FieldSelector: fieldSelector, TimeoutSeconds: &timeout},
+	)
+	if err != nil {
+		return nil, false, err
+	}
+
+	resourceVersion := listResult.GetResourceVersion()
```

Comment:
Does `client-go` work the same as `kubectl` on this list? When I do something like a `get pods -o yaml` on a cluster (even with the same kind of field selector), I get something back like:
```yaml
apiVersion: v1
items:
  ...
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

So I kind of feel like the resourceVersion here is the version of the list, instead of the version of the actual resource.

Could this function use a `Get` instead of a `List`? And handle the "not found" case more specifically?

Reply:
The `resourceVersion` is from the list, which is intentional since you can't watch a single object directly. You must watch a list with a field selector that filters on the name.

Reply:
It's `kubectl` (at least, my version) working differently. If I get verbose output from it like `get pods -o yaml -v=8`, I can see this in the actual API server response:
```
kind: PodList
metadata:
  resourceVersion: "2177620975"
...
```
but for some reason it doesn't give me that in the "normal" output.